### PR TITLE
Datasets page fix

### PR DESCRIPTION
**Pull request by Ambradibi**
- scrolling issue 
- big gray space on the bottom 
- "add new dataset" doesn't work
- fix the double click problem 
- Allow more room for the title

### CODE REVIEW

**Review by cbotsikas — CHANGES_REQUESTED** [review]
This needs more work.  
- The panel that has the README.md is not there anymore and is not resizable.
- 'big gray space on the bottom' still exists. Did you just made it a bit smaller?
- 'double click' issue is not fixed either, use a project with many datasets (eg, from Admin -> Road) scroll down, try and access any dataset. I think the main issue is that the panel is resizing before the user has the chance to doubleclick. You can't have both click and dblclick execute the same time.
- for double click, i can see that you added `ng-dblclick="projectCtrl.browseDataset(dataset)"`. What about the code that existed? Don't you need to remove the old code that opened the dataset?
- When you select a dataset, the icon changes to open ONLY for the first dataset. Selecting any other, doesn't change the icon.
image

**Comment by Ambradibi** [comment]
@cbotsikas 

**The panel that has the README.md is not there anymore and is not resizable:** 
yes because if you want to make the part of the datasets scrollable I had to remove the bar that resized the readme because it is anchored to the part above. I found this solution that still shows all the data.

**'big gray space on the bottom' still exists. Did you just made it a bit smaller?:**
I have decreased it. Do you resize it even more?

**'double click' issue is not fixed either, use a project with many datasets (eg, from Admin -> Road) scroll down, try and access any dataset. I think the main issue is that the panel is resizing before the user has the chance to doubleclick. You can't have both click and dblclick execute the same time.** 
if it is not possible to have both a click and double click what solution do you suggest? Is it not better to divide the clickable areas?

**for double click, i can see that you added ng-dblclick="projectCtrl.browseDataset(dataset)". What about the code that existed? Don't you need to remove the old code that opened the dataset?:**
before if I remember correctly there was not a double click but a single click on two different areas of the dataset.

**When you select a dataset, the icon changes to open ONLY for the first dataset. Selecting any other, doesn't change the icon:**
I had already noticed this problem and was solving it. The problem is that it does not remove the class after the first click.

**Comment by cbotsikas** [comment]
> **The panel that has the README.md is not there anymore and is not resizable:**
> yes because if you want to make the part of the datasets scrollable I had to remove the bar that resized the readme because it is anchored to the part above. I found this solution that still shows all the data.

This is a wrong solution. Have a look on how it should be done and compare with the original hopswork-web ui. You can test it [for example here](https://bbc6.sics.se:8181/hopsworks/#!/project/1049/datasets)

> **'big gray space on the bottom' still exists. Did you just made it a bit smaller?:**
> I have decreased it. Do you resize it even more?

I would expect that the content in the grey box would be centered, having the same space from the edges. [Here's an example](https://bbc6.sics.se:8181/hopsworks/#!/project/1056/datasets) from hopswork-web ui.

> **'double click' issue is not fixed either, use a project with many datasets (eg, from Admin -> Road) scroll down, try and access any dataset. I think the main issue is that the panel is resizing before the user has the chance to doubleclick. You can't have both click and dblclick execute the same time.**
> if it is not possible to have both a click and double click what solution do you suggest? Is it not better to divide the clickable areas?

You need to use timeout on your single click so that you can check if the dbclick wasn't executed. I'm sure you can find many solutions, since it's common problem.

> **for double click, i can see that you added ng-dblclick="projectCtrl.browseDataset(dataset)". What about the code that existed? Don't you need to remove the old code that opened the dataset?:**
> before if I remember correctly there was not a double click but a single click on two different areas of the dataset.

How is it working now on dev branch? [Here](https://bbc6.sics.se:8181/aegis/#!/project/1056/datasets), when I double click the icon, it opens the dataset. 

> **When you select a dataset, the icon changes to open ONLY for the first dataset. Selecting any other, doesn't change the icon:**
> I had already noticed this problem and was solving it. The problem is that it does not remove the class after the first click.

Can you fix it? Again, in hopsworks-web ui, it works

**Comment by Ambradibi** [comment]
@cbotsikas today I work on these changes

**Comment by Ambradibi** [comment]
@cbotsikas we should have made all the points. Can you give me feedback please?

File: aegis-web/yo/app/styles/angular-resizable.css [hunk 1]
```diff
@@ -144,15 +144,21 @@
     height: 70%; 
     max-height: 90%; 
     min-height: 30%;
-    overflow: hidden;
-    z-index: 2;
+    overflow-y: scroll;
```
Comment (cbotsikas) [inline aegis-web/yo/app/styles/angular-resizable.css]: I think you don't need that anymore.

File: aegis-web/yo/app/scripts/controllers/datasetsCtrl.js [hunk 2]
```diff
@@ -1247,6 +1246,13 @@ angular.module('hopsWorksApp')
               }
               return "hdfs://" + selectedFile.path;
             };
+            
+            self.singleClick = function() {
+              alert('Single Click');
+            }
+            self.doubleClick = function() {
+              alert('Double Click');
+            }
```
Comment (cbotsikas) [inline aegis-web/yo/app/scripts/controllers/datasetsCtrl.js]: Is this just a debug code? we don't need that commited

**Review by cbotsikas — CHANGES_REQUESTED** [review]
This is looking better now. You need to read again all your changes, I think you have leftovers from your various attempts. 
Since the actual changes are not that many, maybe you should consider starting a new branch from the latest `dev` and manually copy all the necessary changes from here. This will also fix the merge conflict we have now and it wont clatter the git history with all 12 commits.

**Comment by Ambradibi** [comment]
@cbotsikas now I clean everything. It was to see if it was okay. Thanks

**Comment by Ambradibi** [comment]
@cbotsikas I create the new branch now

**Comment by cbotsikas** [comment]
closing since it was included in #61

**Pull request closed without merging**